apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -185,12 +191,45 @@ const saveAiCustomization = async (
 
   // Notify the UI that a save is in progress.
   dispatch(startSave(saveType));
+  Lab2Registry.getInstance()
+    .getMetricsReporter()
+    .incrementCounter('Aichat.SaveStarted');
+
+  // Detect toxicity in the student's customizations
+  const toxicity = await detectToxicityInCustomizations(
+    trimmedCurrentAiCustomizations
+  );
+
+  // If any fields were flagged for toxicity, display a notification and don't try to save.
+  if (toxicity.flaggedFields.length > 0) {
+    // Log for analysis purposes
+    Lab2Registry.getInstance().getMetricsReporter().logInfo({
+      message: 'Toxicity detected in AI customizations',
+      flaggedFields: toxicity.flaggedFields,
+      customizations: trimmedCurrentAiCustomizations,
```

Comment:
AiCustomizations includes model card info which can be a lot of text which is not checked for safety. What do you think about only sending system prompt and retrievals info?

Reply:
Yeah, I had a similar thought - figured it might be useful to have the whole customizations context there for reference? But you're right that the model card info is not relevant. I think stripping it down to retrieval and system prompt makes sense.

Reply:
Updated this to only log system prompt and retrieval